src/zino/tasks/bgpstatemonitortask.py

```diff
@@ -0,0 +1,410 @@
+        bgp_style = await self._get_bgp_style()
+        if bgp_style != self.device_state.bgp_style:
+            _logger.debug(
+                f"Router {self.device_state.name} changed its BGP style from '{self.device_state.bgp_style}' to "
+                f"'{bgp_style}'"
+            )
+            self.device_state.bgp_style = bgp_style
+        if not bgp_style:
+            return
+
+        local_as = await self._get_local_as(bgp_style=bgp_style)
+        uptime = await self._get_uptime()
+        if not local_as or not uptime:
+            return
+
+        if bgp_style == "juniper":
+            bgp_info = await self._get_juniper_bgp_info()
+        elif bgp_style == "cisco":
+            bgp_info = await self._get_cisco_bgp_info()
+        elif bgp_style == "general":
+            bgp_info = await self._get_general_bgp_info()
+
+        if not bgp_info:
+            return
+
+        bgp_info = self._fixup_ip_addresses(bgp_info=bgp_info)
+
+        for oid, result in bgp_info.items():
+            self._update_single_bgp_entry(oid=oid, row=result, local_as=local_as, uptime=uptime)
+
+    async def _get_bgp_style(self) -> Optional[BgpStyle]:
+        snmp = SNMP(self.device)
+
+        if await snmp.subtree_is_supported("BGP4-V2-MIB-JUNIPER", "jnxBgpM2"):
+            return "juniper"
+
+        if await snmp.subtree_is_supported("CISCO-BGP4-MIB", "cbgpPeer2Table"):
+            return "cisco"
+
+        if await snmp.subtree_is_supported("BGP4-MIB", "bgp"):
+            return "general"
+
+        return None
+
+    async def _get_local_as(self, bgp_style: BgpStyle) -> Optional[int]:
+        snmp = SNMP(self.device)
+        if bgp_style == "juniper":
+            # Juniper has multiple entries for this, so we just take the first
+            juniper_local_as = await snmp.getnext("BGP4-V2-MIB-JUNIPER", "jnxBgpM2PeerLocalAs")
+            if snmp._is_in_scope(juniper_local_as, ("BGP4-V2-MIB-JUNIPER", "jnxBgpM2PeerLocalAs")):
+                return juniper_local_as.value
+            _logger.info(f"router {self.device.name} misses jnxBgpM2PeerLocalAs")
+            return None
+        if bgp_style == "cisco":
+            try:
+                return (await snmp.get("CISCO-BGP4-MIB", "cbgpLocalAs", 0)).value
+            except NoSuchNameError:
+                _logger.info(f"router {self.device.name} misses cbgpLocalAs")
+                return None
+        if bgp_style == "general":
+            try:
+                return (await snmp.get("BGP4-MIB", "bgpLocalAs", 0)).value
+            except NoSuchNameError:
+                _logger.info(f"router {self.device.name} misses bgpLocalAs")
+                return None
+
+    async def _get_uptime(self) -> Optional[int]:
+        snmp = SNMP(self.device)
+        try:
+            return (await snmp.get("SNMPv2-MIB", "sysUpTime", 0)).value / 100
+        except NoSuchNameError:
+            _logger.info(f"router {self.device.name} misses sysUpTime")
+            return None
```

Comment:
Given a constant defined at the module-level as

```python
LOCAL_AS_OBJECTS = {
    "juniper": ("BGP4-V2-MIB-JUNIPER", "jnxBgpM2PeerLocalAs"),
    "cisco": ("CISCO-BGP4-MIB", "cbgpLocalAs"),
    "general": ("BGP4-MIB", "bgpLocalAs")
}
```

This method could be simplified as:


```suggestion
    async def _get_local_as(self, bgp_style: BgpStyle) -> Optional[int]:
        snmp = SNMP(self.device)
        try:
            mib, object_name = LOCAL_AS_OBJECTS[bgp_style]
        except KeyError:
            return

        response = await snmp.getnext(mib, object_name)
        if snmp._is_in_scope(response, (mib, object_name)):
            return response.value
        else:
            _logger.info(f"router {self.device.name} misses {object_name}")
```